server/queries.js

```diff
@@ -33,12 +33,21 @@ const getNameById = async (request, response) => {
 
 const createName = async (request, response) => {
   const { name, is_male } = request.body;
-  const result = await knex('pet_names').insert({ name, is_male }, [
-    'id',
-    'name',
-    'is_male',
-  ]);
-  response.status(201).json(result[0]);
+  try {
+    const result = await knex('pet_names').insert({ name, is_male }, [
+      'id',
+      'name',
+      'is_male',
+    ]);
+    response.status(201).json(result[0]);
+  } catch (error) {
+    console.log(error);
+    if (error.code == '23505') {
+      response.status(598).json({ msg: `${name} already exists` });
```

Comment:
Why 598? This appears to be a network timeout code. https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#598

I usually see 422 used as the response status for validation issues. Any chance of using that or are there any issues with that?

Reply:
changed to 422.  I just picked a random response status code for testing.